_posts/2021-01-18-getting-started-kafka.adoc

```diff
@@ -0,0 +1,303 @@
+    }
+}
+----
+
+== Step 7 - Let's get this running!
+
+Well, first, we need a Kafka broker.
+You can follow the https://kafka.apache.org/quickstart[Apache Kafka quickstart], or use the following `docker-compose.yaml` file:
+
+[source, yaml]
+----
+version: '2'
+
+services:
+
+ zookeeper:
+ image: strimzi/kafka:0.19.0-kafka-2.5.0
+ command: [
+ "sh", "-c",
+ "bin/zookeeper-server-start.sh config/zookeeper.properties"
+ ]
+ ports:
+ - "2181:2181"
+ environment:
+ LOG_DIR: /tmp/logs
+
+ kafka:
+ image: strimzi/kafka:0.19.0-kafka-2.5.0
+ command: [
+ "sh", "-c",
+ "bin/kafka-server-start.sh config/server.properties --override listeners=$${KAFKA_LISTENERS} --override advertised.listeners=$${KAFKA_ADVERTISED_LISTENERS} --override zookeeper.connect=$${KAFKA_ZOOKEEPER_CONNECT}"
+ ]
+ depends_on:
+ - zookeeper
+ ports:
+ - "9092:9092"
+ environment:
+ LOG_DIR: "/tmp/logs"
+ KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://localhost:9092
+ KAFKA_LISTENERS: PLAINTEXT://0.0.0.0:9092
+ KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
+----
+
+Copy the `docker-compose.yaml` file in your project, and from a terminal, start your broker with: `docker-compose up -d`
+
+Then, run the application using:
+
+[source, bash]
+----
+./mvnw quarkus:dev
+----
+
+The application runs in dev mode, meaning that you can still update the code.
+It will reload it automatically.
+
+In another terminal, emit a few HTTP POST request such as:
+
+[source, bash]
+----
+curl --header "Content-Type: application/json" \
+--request POST \
+--data '{"year":1994, "title":"The Shawshank Redemption"}' \
+http://localhost:8080/
+
+curl --header "Content-Type: application/json" \
+--request POST \
+--data '{"year":1972, "title":"The Godfather"}' \
+http://localhost:8080/
+
+curl --header "Content-Type: application/json" \
+--request POST \
+--data '{"year":2008, "title":"The Dark Knight"}' \
+http://localhost:8080/
+
+curl --header "Content-Type: application/json" \
+--request POST \
+--data '{"year":1994, "title":"Pulp Fiction"}' \
+http://localhost:8080/
+
+curl --header "Content-Type: application/json" \
+--request POST \
+--data '{"year":2010, "title":"Inception"}' \
+http://localhost:8080/
+----
+
+In the terminal running the application, you will see:
+
+[source, text]
+----
+...
+2021-01-13 09:29:41,087 INFO  [org.acm.MovieConsumer] (vert.x-eventloop-thread-9) Got a movie: 1994 - Pulp Fiction
+2021-01-13 09:29:41,114 INFO  [org.acm.MovieConsumer] (vert.x-eventloop-thread-9) Got a movie: 2010 - Inception
+...
+----
+
+It works!
+
+== Step 8 - Native packaging
+
+If you have https://quarkus.io/guides/building-native-image[GraalVM installed and configured correctly], you can package this application as a native executable:
+
+[source, bash]
+----
+./mvnw package -Pnative
+----
+
+Then, execute your native executable with: `./target/getting-started-kafka-1.0.0-SNAPSHOT-runner`, and you get a Quarkus application using Kafka starting in a few milliseconds and consuming a ridiculous amount of memory.
```

Comment:
Do you need to mention the memory is ridiculously "small" rather than leaving it open to interpretation?

Reply:
I've added the measure: 30mb!